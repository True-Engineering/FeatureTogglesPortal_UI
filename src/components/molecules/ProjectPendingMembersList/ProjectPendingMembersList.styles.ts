import {
  commonTheme,
  ComponentStyles,
} from '@true-engineering/true-react-common-ui-kit';

const { colors } = commonTheme;

export const styles = {
  root: {},

  nameColumn: {
    width: 256,
    flexShrink: 0,
    paddingLeft: 12,
  },

  emailColumn: {
    flexGrow: 1,
  },

  actionsColumn: {
    width: 256,
    flexShrink: 0,
    justifyContent: 'center',
  },

  nameCell: {
    display: 'flex',
    gap: 12,
    padding: [16, 12],
  },

  userName: {
    extend: 'ellipsis',

    color: colors.FONT_MAIN,
    fontSize: 14,
    lineHeight: '18px',
    letterSpacing: '0.15px',
  },

  email: {
    extend: 'ellipsis',
    color: colors.FONT_LABEL,
    fontSize: 14,
    lineHeight: '18px',
    letterSpacing: '0.15px',
  },

  actionsCell: {
    display: 'flex',
    justifyContent: 'center',
    gap: 12,
  },

  ellipsis: {
    display: 'inline-block',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
  },

  empty: {
    color: colors.FONT_LABEL,
    textAlign: 'center',
  },
};

export const selectTweakStyles = {
  input: {
    inputWrapper: {
      border: 'none',
      height: 32,
    },
  },

  arrow: {
    top: 4,
    right: 16,
  },
};

export type ProjectPendingMembersListStyles = ComponentStyles<typeof styles>;
