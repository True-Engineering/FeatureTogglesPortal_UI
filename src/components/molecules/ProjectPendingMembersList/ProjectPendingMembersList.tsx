import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import clsx from 'clsx';
import { Button } from '@true-engineering/true-react-common-ui-kit';
import { AccountAvatar } from '../../atoms';
import { useFeatureTogglesTheme } from '../../../hooks';
import { IProject, IProjectId, IProjectMember, IUser } from '../../../types';
import { tableStyles } from '../../commonStyles';
import { styles } from './ProjectPendingMembersList.styles';

export interface IProjectPendingMembersListProps {
  users: IProjectMember[];
  onAcceptUser: (args: {
    user: IUser;
    projectName: IProject['name'];
    projectId: IProjectId;
  }) => void;
  onRejectUser: (args: {
    user: IUser;
    projectName: IProject['name'];
    projectId: IProjectId;
  }) => void;
}

const ProjectPendingMembersList: FC<IProjectPendingMembersListProps> = ({
  users,
  onAcceptUser,
  onRejectUser,
}) => {
  const { classes: tableClasses } = useFeatureTogglesTheme(
    'tableStyles',
    tableStyles,
    {},
  );

  const { classes } = useFeatureTogglesTheme(
    'ProjectPendingMembersList',
    styles,
  );
  const { t } = useTranslation();

  if (users.length === 0) {
    return <div className={classes.empty}>{t('noMembers')}</div>;
  }

  return (
    <div className={clsx(tableClasses.table, classes.root)}>
      {users.map(({ user, projectId, projectName }) => {
        const { id: userId, userName, avatarUrl, email } = user;
        return (
          <div key={userId} className={tableClasses.row}>
            <div
              className={clsx(
                tableClasses.cell,
                classes.nameColumn,
                classes.nameCell,
              )}
            >
              <AccountAvatar name={userName} avatarUrl={avatarUrl} />
              <div className={classes.userName} title={userName}>
                {userName}
              </div>
            </div>
            <div className={clsx(tableClasses.cell, classes.emailColumn)}>
              <div className={classes.email} title={email}>
                {email}
              </div>
            </div>
            <div className={clsx(tableClasses.cell, classes.actionsColumn)}>
              <div className={classes.actionsCell}>
                <Button
                  onClick={() => onRejectUser({ user, projectId, projectName })}
                  view="secondary"
                  size="s"
                >
                  {t('reject')}
                </Button>
                <Button
                  onClick={() => onAcceptUser({ user, projectId, projectName })}
                  view="primary"
                  size="s"
                >
                  {t('accept')}
                </Button>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default ProjectPendingMembersList;
