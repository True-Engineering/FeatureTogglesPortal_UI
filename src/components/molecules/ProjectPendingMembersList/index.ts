export { default as ProjectPendingMembersList } from './ProjectPendingMembersList';
export * from './ProjectPendingMembersList';

export type { ProjectPendingMembersListStyles } from './ProjectPendingMembersList.styles';
