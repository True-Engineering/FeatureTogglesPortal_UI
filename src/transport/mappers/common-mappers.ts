import { parseISO } from 'date-fns';
import {
  IFlag,
  IFlagChanges,
  IHistoryOfChangesListItem,
  IOrganization,
  IProject,
  IProjectEnvironment,
  IProjectEnvironmentInstance,
} from '../../types';
import {
  IRawFlag,
  IRawFlagChanges,
  IRawHistoryOfChangesListItem,
  IRawOrganization,
  IRawProject,
  IRawProjectEnvironment,
} from './types';
import { mapperFlippingStrategyFromApi } from './mapper-flipping-strategy';

export const trimStringsInObject = <T extends Record<string, any>>(obj: T): T =>
  Object.fromEntries(
    Object.entries(obj).map(([k, v]) => {
      if (typeof v === 'object' && !Array.isArray(v) && v !== null) {
        return [k, trimStringsInObject(v)];
      }
      if (typeof v === 'string') {
        return [k, v.trim()];
      }
      return [k, v];
    }),
  ) as T;

// 'true' -> true в остальных случаях 'value' -> false
const stringToBoolean = (value: string): boolean => value === 'true';

export const mapperOrganization = ({
  projects,
  ...rest
}: IRawOrganization): IOrganization<Date> => ({
  ...rest,
  projects: projects.map(mapperProject),
});

export const mapperProject = ({
  environments,
  ...rest
}: IRawProject): IProject<Date> => ({
  ...rest,
  environments: environments.map(mapperProjectEnvironment),
});

export const mapperProjectEnvironment = ({
  instances,
  properties,
  emails,
  ...rest
}: IRawProjectEnvironment): IProjectEnvironment<Date> => ({
  ...rest,
  instances: instances.map(mapperEnvironmentInstance),
  emails: emails ?? [],
  freezingEnable:
    properties?.FREEZING_ENABLE !== undefined
      ? stringToBoolean(properties?.FREEZING_ENABLE)
      : false,
  freezingUser: properties?.FREEZING_USER,
  freezingEndTime: properties?.FREEZING_END_TIME
    ? new Date(properties?.FREEZING_END_TIME)
    : undefined,
});

export const mapperEnvironmentInstance = ({
  updated,
  ...rest
}: IProjectEnvironmentInstance<string>): IProjectEnvironmentInstance<Date> => ({
  ...rest,
  updated: updated === undefined ? undefined : parseISO(updated),
});

export const mapperFlag = ({
  uid,
  description,
  type,
  tags,
  sprint,
  group,
  environments,
}: IRawFlag): IFlag => ({
  name: uid,
  description: description ?? '',
  sprint: sprint ?? '',
  type,
  tag: (tags ?? [])[0],
  group: group ?? '',
  environments: (environments ?? []).map(
    ({ flippingStrategy, enable, permissions = [], ...restEnv }) => ({
      ...restEnv,
      permissions,
      isEnabled: enable,
      flippingStrategy:
        flippingStrategy === undefined
          ? undefined
          : mapperFlippingStrategyFromApi(flippingStrategy),
    }),
  ),
});

const mapperChanges = ({
  uid,
  environments,
  tags,
  ...rest
}: IRawFlagChanges): IFlagChanges => ({
  tag:
    tags !== undefined
      ? {
          old: (tags.old ?? []).join(', '),
          new: (tags.new ?? []).join(', '),
        }
      : undefined,
  ...rest,
});

export const historyOfChangesListItemMapper = ({
  id,
  action,
  userName,
  featureFlag,
  environment,
  time,
  changes,
  creationInfo,
}: IRawHistoryOfChangesListItem): IHistoryOfChangesListItem => ({
  id,
  action,
  userName,
  featureFlag: {
    ...mapperFlag(featureFlag),
    tag: (featureFlag.tags ?? []).join(', '),
  },
  environment,
  time: new Date(time),
  changes: changes !== undefined ? mapperChanges(changes) : {},
  creationInfo:
    creationInfo !== undefined
      ? {
          ...mapperFlag(creationInfo),
          tag: (creationInfo.tags ?? []).join(', '),
        }
      : undefined,
});
