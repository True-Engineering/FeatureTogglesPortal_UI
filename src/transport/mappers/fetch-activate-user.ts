import { IRequestParams } from '@true-engineering/true-react-common-api-client';
import { IProjectId, IUserId } from '../../types';

export interface IFetchActivateUserRequestParams extends IRequestParams {
  userId: IUserId;
  projectId: IProjectId;
}
