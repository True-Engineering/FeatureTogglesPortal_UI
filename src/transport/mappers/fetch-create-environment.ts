import { IRequestParams } from '@true-engineering/true-react-common-api-client';
import { IProjectEnvironment, IProjectId } from '../../types';
import { IRawProjectEnvironment } from './types';
import {
  mapperProjectEnvironment,
  trimStringsInObject,
} from './common-mappers';

export interface IFetchCreateEnvironmentRequestParams extends IRequestParams {
  projectId: IProjectId;
}

export interface IFetchCreateEnvironment {
  name: string;
}

export type IFetchCreateEnvironmentResponse = IProjectEnvironment;

export type IFetchCreateEnvironmentResponseRaw = IRawProjectEnvironment;

export const fetchCreateEnvironment = mapperProjectEnvironment;

export const fetchCreateEnvironmentRequestMapper = (
  body: IFetchCreateEnvironment,
) => trimStringsInObject(body);
