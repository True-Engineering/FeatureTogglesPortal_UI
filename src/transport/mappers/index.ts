import {
  fetchCreateEnvironment,
  fetchCreateEnvironmentRequestMapper,
} from './fetch-create-environment';
import {
  fetchCreateFlagRequestMapper,
  fetchCreateFlag,
} from './fetch-create-flag';
import {
  fetchCreateProject,
  fetchCreateProjectRequestMapper,
} from './fetch-create-project';
import {
  fetchEditEnvironment,
  fetchEditEnvironmentRequestMapper,
} from './fetch-edit-environment';
import { fetchEditFlagRequestMapper, fetchEditFlag } from './fetch-edit-flag';
import {
  fetchEditFlagStrategyRequestMapper,
  fetchEditFlagStrategy,
} from './fetch-edit-flag-strategy';
import { fetchEditMemberEnvironmentPermission } from './fetch-edit-member-environment-permission';
import { fetchEditMemberProjectPermission } from './fetch-edit-member-project-permission';
import {
  fetchEditProject,
  fetchEditProjectRequestMapper,
} from './fetch-edit-project';
import { fetchFreezeFeatureFlagsRequestParamsMapper } from './fetch-freeze-feature-flags';
import { fetchGetCompareEnvironmentsState } from './fetch-get-compare-environment-state';
import { fetchGetEnvironment } from './fetch-get-environment';
import { fetchGetFeatureFlagsForFilters } from './fetch-get-feature-flags-for-filters';
import {
  fetchGetHistoryOfChanges,
  fetchGetHistoryOfChangesParamsMapper,
} from './fetch-get-history-of-changes';
import { fetchGetOrganizationMembers } from './fetch-get-organization-members';
import { fetchGetProject } from './fetch-get-project';
import { fetchGetProjectMembers } from './fetch-get-project-members';
import { fetchGetTagsForFilters } from './fetch-get-tags-for-filters';
import { fetchGetUser } from './fetch-get-user';
import { fetchGetUsersForFilters } from './fetch-get-users-for-filters';
import { fetchLoadEnvironments } from './fetch-load-environments';
import { fetchLoadFlags } from './fetch-load-flags';
import { fetchLoadOrganizations } from './fetch-load-organizations';
import { fetchLoadProjects } from './fetch-load-projects';
import { fetchSyncEnvironmentsInfo } from './fetch-sync-environments-info';
import { fetchSyncEnvironmentsUpload } from './fetch-sync-environments-upload';
import { fetchSyncPortalsInfo } from './fetch-sync-portals-info';
import { fetchSyncPortalsUpload } from './fetch-sync-portals-upload';
import { fetchUpdateEnvironmentNotificationsBodyRequest } from './fetch-update-email-notifications';

const mappers = {
  fetchGetUser,
  fetchLoadFlags,
  fetchCreateFlagRequestMapper,
  fetchCreateFlag,
  fetchEditFlagRequestMapper,
  fetchEditFlag,
  fetchEditFlagStrategyRequestMapper,
  fetchEditFlagStrategy,
  fetchCreateEnvironmentRequestMapper,
  fetchCreateEnvironment,
  fetchCreateProjectRequestMapper,
  fetchCreateProject,
  fetchEditEnvironmentRequestMapper,
  fetchEditEnvironment,
  fetchEditProjectRequestMapper,
  fetchEditProject,
  fetchGetEnvironment,
  fetchGetProject,
  fetchLoadEnvironments,
  fetchLoadOrganizations,
  fetchLoadProjects,
  fetchGetOrganizationMembers,
  fetchGetProjectMembers,
  fetchEditMemberProjectPermission,
  fetchEditMemberEnvironmentPermission,
  fetchSyncPortalsInfo,
  fetchSyncPortalsUpload,
  fetchSyncEnvironmentsInfo,
  fetchSyncEnvironmentsUpload,
  fetchGetCompareEnvironmentsState,
  fetchUpdateEnvironmentNotificationsBodyRequest,
  fetchGetHistoryOfChanges,
  fetchGetFeatureFlagsForFilters,
  fetchGetUsersForFilters,
  fetchGetHistoryOfChangesParamsMapper,
  fetchGetTagsForFilters,
  fetchFreezeFeatureFlagsRequestParamsMapper,
};

export default mappers;

export * from './mapper-flipping-strategy';
export * from './fetch-load-flags';
export * from './fetch-create-flag';
export * from './fetch-edit-flag';
export * from './fetch-edit-flag-strategy';
export * from './fetch-get-user';
export * from './fetch-enable-flag';
export * from './fetch-delete-flag';
export * from './fetch-download-flags-file';
export * from './fetch-upload-flags-file';
export * from './fetch-load-organizations';
export * from './fetch-load-projects';
export * from './fetch-create-project';
export * from './fetch-get-project';
export * from './fetch-edit-project';
export * from './fetch-mark-default-project';
export * from './fetch-delete-project';
export * from './fetch-load-environments';
export * from './fetch-create-environment';
export * from './fetch-get-environment';
export * from './fetch-edit-environment';
export * from './fetch-delete-environment';
export * from './fetch-update-token';
export * from './fetch-delete-environment-instance';
export * from './fetch-invite-link-for-project';
export * from './fetch-invite';
export * from './fetch-get-organization-members';
export * from './fetch-get-project-members';
export * from './fetch-edit-member-project-permission';
export * from './fetch-edit-member-environment-permission';
export * from './fetch-delete-organization-member';
export * from './fetch-delete-project-member';
export * from './fetch-activate-user';
export * from './fetch-sync-portals-info';
export * from './fetch-sync-portals-upload';
export * from './fetch-sync-environments-info';
export * from './fetch-sync-environments-upload';
export * from './fetch-get-compare-environment-state';
export * from './fetch-update-email-notifications';
export * from './fetch-update-flags-state';
export * from './fetch-get-history-of-changes';
export * from './fetch-get-feature-flags-for-filters';
export * from './fetch-get-users-for-filters';
export * from './fetch-get-tags-for-filters';
export * from './fetch-freeze-feature-flags';
export * from './fetch-unfreeze-feature-flags';
