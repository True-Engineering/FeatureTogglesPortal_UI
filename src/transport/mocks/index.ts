import {
  IFetchCreateProjectResponseRaw,
  IFetchEditProjectResponseRaw,
  IFetchGetProjectResponseRaw,
  IFetchGetUserResponseRaw,
  IFetchLoadFlagsResponseRaw,
  IFetchLoadOrganizationsResponseRaw,
  IFetchLoadProjectsResponseRaw,
  IFetchLoadEnvironmentsResponseRaw,
  IFetchCreateEnvironmentResponseRaw,
  IFetchGetEnvironmentResponseRaw,
  IFetchEditEnvironmentResponseRaw,
  IFetchUpdateEnvironmentTokenResponseRaw,
  IFetchInviteLinkForProjectResponseRaw,
  IFetchGetOrganizationMembersResponseRaw,
  IFetchGetProjectMembersResponseRaw,
  IFetchCreateFlagResponseRaw,
  IFetchEditFlagResponseRaw,
  IFetchSyncPortalsInfoResponseRaw,
  IFetchSyncEnvironmentsInfoResponseRaw,
  IFetchSyncPortalsUploadResponseRaw,
  IFetchGetCompareEnvironmentsStateResponseRaw,
  IFetchFreezeFeatureFlagsResponseRaw,
  IFetchUnfreezeFeatureFlagsResponseRaw,
} from '../mappers';
import fetchEditMemberEnvironmentPermission from './fetch-edit-member-environment-permission';
import fetchEditMemberProjectPermission from './fetch-edit-member-project-permission';
import fetchGetFeatureFlagsForFilters from './fetch-get-feature-flags-for-filters';
import fetchGetHistoryOfChanges from './fetch-get-history-of-changes';
import fetchGetTagsForFilters from './fetch-get-tags-for-filters';
import fetchGetUsersForFilters from './fetch-get-users-for-filters';
import {
  flags,
  user,
  project,
  projectEnvironment,
  organizations,
  environmentToken,
  inviteUuid,
  projectMembers,
  organizationMembers,
  syncPortalsInfo,
  syncEnvironmentsInfo,
  compareEnvironmentsState,
} from './fixtures';
import { mockTemplate } from './mock-template';

const mocks = {
  getUser: mockTemplate<IFetchGetUserResponseRaw>('getUser', user),
  // FLAGS
  createFlag: mockTemplate<IFetchCreateFlagResponseRaw>('createFlag', flags[0]),
  editFlag: mockTemplate<IFetchCreateFlagResponseRaw>('editFlag', flags[0]),
  editFlagStrategy: mockTemplate<IFetchEditFlagResponseRaw>(
    'editFlagStrategy',
    flags[0],
  ),
  loadFlags: mockTemplate<IFetchLoadFlagsResponseRaw>('loadFlags', flags),
  enableFlag: mockTemplate<boolean>('enableFlag', true),
  disableFlag: mockTemplate<boolean>('disableFlag', true),
  deleteFlag: mockTemplate<boolean>('deleteFlag', true),
  downloadFlagsFile: mockTemplate<unknown>('downloadFlagsFile', flags),
  uploadFlagsFile: mockTemplate<boolean>('uploadFlagsFile', true),

  syncPortalsInfo: mockTemplate<IFetchSyncPortalsInfoResponseRaw>(
    'syncPortalsInfo',
    syncPortalsInfo,
  ),
  syncPortalsUpload: mockTemplate<IFetchSyncPortalsUploadResponseRaw>(
    'syncPortalsUpload',
    flags,
  ),
  syncEnvironmentsInfo: mockTemplate<IFetchSyncEnvironmentsInfoResponseRaw>(
    'syncEnvironmentsInfo',
    syncEnvironmentsInfo,
  ),
  syncEnvironmentsUpload: mockTemplate<IFetchSyncPortalsUploadResponseRaw>(
    'syncEnvironmentsUpload',
    flags,
  ),

  // ORGANIZATION
  loadOrganizations: mockTemplate<IFetchLoadOrganizationsResponseRaw>(
    'loadOrganizations',
    organizations,
  ),

  // PROJECT
  loadProjects: mockTemplate<IFetchLoadProjectsResponseRaw>(
    'loadProjects',
    organizations[0].projects,
  ),
  createProject: mockTemplate<IFetchCreateProjectResponseRaw>(
    'createProject',
    project,
  ),
  getProject: mockTemplate<IFetchGetProjectResponseRaw>('getProject', project),
  editProject: mockTemplate<IFetchEditProjectResponseRaw>(
    'editProject',
    project,
  ),
  markDefaultProject: mockTemplate<boolean>('markDefaultProject', true),
  deleteProject: mockTemplate<boolean>('deleteProject', true),

  // MEMBERS
  getInviteLinkForProject: mockTemplate<IFetchInviteLinkForProjectResponseRaw>(
    'getInviteLinkForProject',
    inviteUuid,
  ),
  fetchInvite: mockTemplate<boolean>('fetchInvite', true),
  getOrganizationMembers: mockTemplate<IFetchGetOrganizationMembersResponseRaw>(
    'getOrganizationMembers',
    organizationMembers,
  ),
  getProjectMembers: mockTemplate<IFetchGetProjectMembersResponseRaw>(
    'getProjectMembers',
    projectMembers,
  ),
  editMemberProjectPermission: fetchEditMemberProjectPermission,
  editMemberEnvironmentPermission: fetchEditMemberEnvironmentPermission,
  activateUser: mockTemplate<boolean>('activateUser', true),
  deleteOrganizationMember: mockTemplate<boolean>(
    'deleteOrganizationMember',
    true,
  ),
  deleteProjectMember: mockTemplate<boolean>('deleteProjectMember', true),

  // ENVIRONMENT
  loadEnvironments: mockTemplate<IFetchLoadEnvironmentsResponseRaw>(
    'loadEnvironments',
    organizations[0]?.projects[0]?.environments,
  ),
  createEnvironment: mockTemplate<IFetchCreateEnvironmentResponseRaw>(
    'createEnvironment',
    projectEnvironment,
  ),
  getEnvironment: mockTemplate<IFetchGetEnvironmentResponseRaw>(
    'getEnvironment',
    projectEnvironment,
  ),
  editEnvironment: mockTemplate<IFetchEditEnvironmentResponseRaw>(
    'editEnvironment',
    projectEnvironment,
  ),
  deleteEnvironment: mockTemplate<boolean>('deleteEnvironment', true),
  updateEnvironmentToken: mockTemplate<IFetchUpdateEnvironmentTokenResponseRaw>(
    'updateEnvironmentToken',
    environmentToken,
  ),

  deleteEnvironmentInstance: mockTemplate<boolean>(
    'deleteEnvironmentInstance',
    true,
  ),

  updateEnvironmentNotifications: mockTemplate<null>(
    'updateEnvironmentNotifications',
    null,
  ),

  getCompareEnvironmentsState: mockTemplate<IFetchGetCompareEnvironmentsStateResponseRaw>(
    'getCompareEnvironmentsState',
    compareEnvironmentsState,
  ),

  updateFlagsState: mockTemplate<null>('updateFlagsState', null),

  getHistoryOfChanges: fetchGetHistoryOfChanges,

  getFeatureFlagsForFilters: fetchGetFeatureFlagsForFilters,

  getUsersForFilters: fetchGetUsersForFilters,

  getTagsForFilters: fetchGetTagsForFilters,

  freezeFeatureFlags: mockTemplate<IFetchFreezeFeatureFlagsResponseRaw>(
    'freezeFeatureFlags',
    organizations[0]?.projects[0]?.environments[0],
  ),

  unfreezeFeatureFlags: mockTemplate<IFetchUnfreezeFeatureFlagsResponseRaw>(
    'unfreezeFeatureFlags',
    organizations[0]?.projects[0]?.environments[0],
  ),
};

export default mocks;
