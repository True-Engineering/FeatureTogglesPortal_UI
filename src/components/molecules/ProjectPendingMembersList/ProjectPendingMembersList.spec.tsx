/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { nullToUndef } from '@true-engineering/true-react-api-client';
import mappers from '../../../transport/mappers';
import { projectMembers as fixture } from '../../../transport/mocks/fixtures';
import ProjectPendingMembersList from './ProjectPendingMembersList';

const allUsers = mappers.fetchGetProjectMembers(nullToUndef(fixture));
const pendingUsers = allUsers.filter(({ user }) => user.status === 'PENDING');

describe('ProjectPendingList рендерится', () => {
  test('Рендер на мок данных', () => {
    const { container } = render(
      <ProjectPendingMembersList
        users={pendingUsers}
        onAcceptUser={() => {}}
        onRejectUser={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('ProjectPendingList корректно отрабатывает сценарии', () => {
  test('Вызывается onAcceptUser по клику на кнопку "Принять"', async () => {
    const onAcceptUser = jest.fn();
    const { findAllByText } = render(
      <ProjectPendingMembersList
        users={pendingUsers}
        onAcceptUser={onAcceptUser}
        onRejectUser={() => {}}
      />,
    );

    const acceptButton = await findAllByText('Принять');
    fireEvent.click(acceptButton[0]);

    expect(onAcceptUser).toBeCalledTimes(1);
  });

  test('Вызывается onRejectUser по клику на кнопку "Отклонить"', async () => {
    const onRejectUser = jest.fn();
    const { findAllByText } = render(
      <ProjectPendingMembersList
        users={pendingUsers}
        onAcceptUser={() => {}}
        onRejectUser={onRejectUser}
      />,
    );

    const acceptButton = await findAllByText('Отклонить');
    fireEvent.click(acceptButton[0]);

    expect(onRejectUser).toBeCalledTimes(1);
  });
});
