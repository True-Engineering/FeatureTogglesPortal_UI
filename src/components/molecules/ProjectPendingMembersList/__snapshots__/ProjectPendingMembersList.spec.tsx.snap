// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPendingList рендерится Рендер на мок данных 1`] = `
<div>
  <div
    class="table root"
  >
    <div
      class="row"
    >
      <div
        class="cell nameColumn nameCell"
      >
        <img
          alt="user-avatar"
          class="imageAvatar"
          src="https://www.gravatar.com/avatar/6786871113358.jpg?d=identicon"
        />
        <div
          class="userName"
          title="Кент Кларк"
        >
          Кент Кларк
        </div>
      </div>
      <div
        class="cell emailColumn"
      >
        <div
          class="email"
          title="kent.clark@metropolis.com"
        >
          kent.clark@metropolis.com
        </div>
      </div>
      <div
        class="cell actionsColumn"
      >
        <div
          class="actionsCell"
        >
          <button
            class="root s cancel"
            type="button"
          >
            <span
              class="content"
            >
              Отклонить
            </span>
          </button>
          <button
            class="root s main"
            type="button"
          >
            <span
              class="content"
            >
              Принять
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell nameColumn nameCell"
      >
        <img
          alt="user-avatar"
          class="imageAvatar"
          src="https://www.gravatar.com/avatar/6786871113358.jpg?d=identicon"
        />
        <div
          class="userName"
          title="Кент Кларк"
        >
          Кент Кларк
        </div>
      </div>
      <div
        class="cell emailColumn"
      >
        <div
          class="email"
          title="kent.clark@metropolis.com"
        >
          kent.clark@metropolis.com
        </div>
      </div>
      <div
        class="cell actionsColumn"
      >
        <div
          class="actionsCell"
        >
          <button
            class="root s cancel"
            type="button"
          >
            <span
              class="content"
            >
              Отклонить
            </span>
          </button>
          <button
            class="root s main"
            type="button"
          >
            <span
              class="content"
            >
              Принять
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell nameColumn nameCell"
      >
        <img
          alt="user-avatar"
          class="imageAvatar"
          src="https://www.gravatar.com/avatar/6786871113358.jpg?d=identicon"
        />
        <div
          class="userName"
          title="Кент Кларк"
        >
          Кент Кларк
        </div>
      </div>
      <div
        class="cell emailColumn"
      >
        <div
          class="email"
          title="kent.clark@metropolis.com"
        >
          kent.clark@metropolis.com
        </div>
      </div>
      <div
        class="cell actionsColumn"
      >
        <div
          class="actionsCell"
        >
          <button
            class="root s cancel"
            type="button"
          >
            <span
              class="content"
            >
              Отклонить
            </span>
          </button>
          <button
            class="root s main"
            type="button"
          >
            <span
              class="content"
            >
              Принять
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell nameColumn nameCell"
      >
        <img
          alt="user-avatar"
          class="imageAvatar"
          src="https://www.gravatar.com/avatar/6786871113358.jpg?d=identicon"
        />
        <div
          class="userName"
          title="Кент Кларк"
        >
          Кент Кларк
        </div>
      </div>
      <div
        class="cell emailColumn"
      >
        <div
          class="email"
          title="kent.clark@metropolis.com"
        >
          kent.clark@metropolis.com
        </div>
      </div>
      <div
        class="cell actionsColumn"
      >
        <div
          class="actionsCell"
        >
          <button
            class="root s cancel"
            type="button"
          >
            <span
              class="content"
            >
              Отклонить
            </span>
          </button>
          <button
            class="root s main"
            type="button"
          >
            <span
              class="content"
            >
              Принять
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="cell nameColumn nameCell"
      >
        <img
          alt="user-avatar"
          class="imageAvatar"
          src="https://www.gravatar.com/avatar/6786871113358.jpg?d=identicon"
        />
        <div
          class="userName"
          title="Кент Кларк"
        >
          Кент Кларк
        </div>
      </div>
      <div
        class="cell emailColumn"
      >
        <div
          class="email"
          title="kent.clark@metropolis.com"
        >
          kent.clark@metropolis.com
        </div>
      </div>
      <div
        class="cell actionsColumn"
      >
        <div
          class="actionsCell"
        >
          <button
            class="root s cancel"
            type="button"
          >
            <span
              class="content"
            >
              Отклонить
            </span>
          </button>
          <button
            class="root s main"
            type="button"
          >
            <span
              class="content"
            >
              Принять
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
